### PR TITLE
docs(Forms): restructure docs

### PR DESCRIPTION
**Pull request by tujoworker**
- [x] Find (or spread them out) a place for [all the demos](https://eufemia.dnb.no/uilib/extensions/forms/demos/).
- **Feature:** Added `level` props to heading components and docs.

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms.mdx [hunk 1]
```diff
@@ -1,20 +1,147 @@
...
+    <Layout.Card spacing="medium">
+      <MyCustomComponent path="/companyName" required />
+    </Layout.Card>
+  </DataContext.Provider>,
+)
+```
+
+### Create your own component – APIs
+
+Eufemia Forms contains of helper fields and tools so you can declaratively create interactive form components that flawlessly integrates between existing data and your custom form components.
```
Comment (henit) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms.mdx:113]:
```suggestion
Eufemia Forms consists of helper fields and tools so you can declaratively create interactive form components that flawlessly integrates between existing data and your custom form components.
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-components/PostalCodeAndCity.mdx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---
+title: 'PostalCodeAndCity'
+description: '`Field.PostalCodeAndCity` is a wrapper component for input of two separate values with user experience tailored for postal code and city values.'
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-components/PostalCodeAndCity.mdx:3]: Not really important, but I think this change also leads to Field.PostalCodeAndCity looking different than the others. Maybe we would add  the same formatting to the description of the other components as well.

Screenshot 2023-09-08 at 09 19 28

**Review by langz — APPROVED** [review]
Looks great, I like that it's a "simpler"/quicker way to see the documentation(less menu entries).

Personally, I think it would be nice for the "general demos" to be even more easily accessed.
https://eufemia-git-docs-forms-restructure-eufemia.vercel.app/uilib/extensions/forms/general-demos/

I think this demo/example is one of the things developers will use/want the most

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 3]
```diff
@@ -51,11 +99,13 @@ export const FeatureFields = () => {
         Visibility,
       }}
     >
-      <Field.FirstName value="John" />
-      <Field.LastName value="Smith" />
-      <Field.NationalIdentityNumber value="20058512345" />
-      <Field.Email value="john@smith.email" />
-      <Field.PhoneNumber value="+47 98765432" />
+      <Layout.Card stack>
+        <Field.String value="John" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:103]:
```suggestion
        <Field.String label="Fornavn" value="John" />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 4]
```diff
@@ -51,11 +99,13 @@ export const FeatureFields = () => {
...
+        <Field.String value="Smith" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:104]:
```suggestion
        <Field.String label="Etternavn" value="Smith" />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 5]
```diff
@@ -78,8 +128,8 @@ export const LayoutComponents = () => {
         <Layout.Card stack>
           <Layout.SubHeading>Name</Layout.SubHeading>
 
-          <Field.FirstName value="John" />
-          <Field.LastName value="Smith" />
+          <Field.String value="John" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:131]:
```suggestion
          <Field.String label="Fornavn" value="John" />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 6]
```diff
@@ -78,8 +128,8 @@ export const LayoutComponents = () => {
...
+          <Field.String value="Smith" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:132]:
```suggestion
          <Field.String label="Etternavn" value="Smith" />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 7]
```diff
@@ -127,8 +177,8 @@ export const VisibilityBasedOnData = () => {
           <Layout.Card stack>
             <Layout.SubHeading>Name</Layout.SubHeading>
 
-            <Field.FirstName path="/firstName" />
-            <Field.LastName path="/lastName" />
+            <Field.String path="/firstName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:180]:
```suggestion
            <Field.String path="/firstName" label="Fornavn" />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 8]
```diff
@@ -127,8 +177,8 @@ export const VisibilityBasedOnData = () => {
...
+            <Field.String path="/lastName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:181]:
```suggestion
            <Field.String path="/lastName" label="Etternavn" />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 9]
```diff
@@ -181,8 +231,8 @@ export const UsingDataContextProvider = () => {
         <Layout.MainHeading>Profile</Layout.MainHeading>
 
         <Layout.Card stack>
-          <Field.FirstName path="/firstName" />
-          <Field.LastName path="/lastName" />
+          <Field.String path="/firstName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:234]:
```suggestion
          <Field.String path="/firstName" / label="Fornavn">
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 10]
```diff
@@ -181,8 +231,8 @@ export const UsingDataContextProvider = () => {
...
+          <Field.String path="/lastName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:235]:
```suggestion
          <Field.String path="/lastName" label="Etternavn"/>
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 11]
```diff
@@ -225,8 +275,8 @@ export const Validation = () => {
         <Layout.MainHeading>Profile</Layout.MainHeading>
 
         <Layout.Card stack>
-          <Field.FirstName path="/firstName" required />
-          <Field.LastName path="/lastName" required />
+          <Field.String path="/firstName" required />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:278]:
```suggestion
          <Field.String path="/firstName" label="Fornavn" required />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 12]
```diff
@@ -225,8 +275,8 @@ export const Validation = () => {
...
+          <Field.String path="/lastName" required />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:279]:
```suggestion
          <Field.String path="/lastName" label="Etternavn" required />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 13]
```diff
@@ -270,8 +320,8 @@ export const WithSteps = () => {
             <Layout.Card stack>
               <Layout.SubHeading>Name</Layout.SubHeading>
 
-              <Field.FirstName path="/firstName" required />
-              <Field.LastName path="/lastName" required />
+              <Field.String path="/firstName" required />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:323]:
```suggestion
              <Field.String path="/firstName" label="Fornavn" required />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 14]
```diff
@@ -270,8 +320,8 @@ export const WithSteps = () => {
...
+              <Field.String path="/lastName" required />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:324]:
```suggestion
              <Field.String path="/lastName" label="Etternavn" required />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 15]
```diff
@@ -301,8 +351,8 @@ export const WithSteps = () => {
 
             <Layout.Card stack>
               <Layout.FlexContainer direction="row">
-                <Value.FirstName path="/firstName" />
-                <Value.LastName path="/lastName" />
+                <Value.String path="/firstName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:354]:
```suggestion
                <Value.String path="/firstName" label="Fornavn"/>
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx [hunk 16]
```diff
@@ -301,8 +351,8 @@ export const WithSteps = () => {
...
+                <Value.String path="/lastName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Examples.tsx:355]:
```suggestion
                <Value.String path="/lastName" label="Etternavn"/>
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx [hunk 17]
```diff
@@ -47,8 +75,8 @@ const handleChange = useCallback((path, value) => {
 
 return (
   <DataContext.Provider data={data} onChange={handleChange}>
-    <Field.FirstName path="/firstName" />
-    <Field.LastName path="/lastName" />
+    <Field.String path="/firstName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx:78]:
```suggestion
    <Field.String path="/firstName" label="Fornavn"/>
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx [hunk 18]
```diff
@@ -47,8 +75,8 @@ const handleChange = useCallback((path, value) => {
...
+    <Field.String path="/lastName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx:79]:
```suggestion
    <Field.String path="/lastName" label="Etternavn"/>
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx [hunk 19]
```diff
@@ -17,11 +45,11 @@ const handleChange = useCallback((path, value) => {
 
 return (
   <div id="my-form">
-    <Field.FirstName
+    <Field.String
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx:48]:
```suggestion
    <Field.String label="Fornavn"
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx [hunk 20]
```diff
@@ -17,11 +45,11 @@ const handleChange = useCallback((path, value) => {
...
       value={data.firstName}
       onChange={(value) => handleChange('firstName', value)}
     />
-    <Field.LastName
+    <Field.String
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx:52]:
```suggestion
    <Field.String label="Etternavn"
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/demo-cases/Examples.tsx [hunk 21]
```diff
@@ -159,8 +159,8 @@ export const BecomeCorporateCustomer = () => {
 
                     <Layout.Card stack>
                       <Layout.FlexContainer direction="row">
-                        <Value.FirstName path="/firstName" />
-                        <Value.LastName path="/lastName" />
+                        <Value.String path="/firstName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/demo-cases/Examples.tsx:162]:
```suggestion
                        <Value.String path="/firstName" label="Fornavn" />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/demo-cases/Examples.tsx [hunk 22]
```diff
@@ -159,8 +159,8 @@ export const BecomeCorporateCustomer = () => {
...
+                        <Value.String path="/lastName" />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/demo-cases/Examples.tsx:163]:
```suggestion
                        <Value.String path="/lastName" label="Etternavn" />
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx [hunk 23]
```diff
@@ -1,8 +1,37 @@
 ---
-showTabs: true
+title: 'DataContext'
+description: 'The `DataContext` interweaves your data-set with your form fields.'
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/DataContext.mdx:3]:
```suggestion
description: '`DataContext` interweaves your data-set with your form fields.'
```

**Review by henit — APPROVED** [review]

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/Field.mdx [hunk 24]
```diff
@@ -1,29 +1,80 @@
 ---
 title: 'Field'
-order: 3
+description: 'Use `Field` for interactive data driven components.'
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/Field.mdx:3]:
```suggestion
description: '`Field` for interactive data driven components.'
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/FieldGroup.mdx [hunk 25]
```diff
@@ -0,0 +1,23 @@
+---
+title: 'FieldGroup'
+description: 'With `FieldGroup` you can group several components and its error states to compose one single component.'
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/create-component/FieldGroup.mdx:3]:
```suggestion
description: '`FieldGroup` you can group several components and its error states to compose one single component.'
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms.mdx [hunk 26]
```diff
@@ -1,20 +1,137 @@
...
+
+Forms is reusable components for data input, data display and surrounding layout for simplified user interface creation in React, built on top of base Eufemia components.
+
+The components constitute flexible building blocks that can be used individually, or together for a more practical data flow. They must also be easy to combine with components from other sources and local functionality in the individual application.
+
+The goal of the Forms extension is to make it easy to build forms and other data input features in DNB applications built with React.
+
+### Additional features
+
+- [Feature components](/uilib/extensions/forms/feature-components) such as Email and PhoneNumber.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms.mdx:40]: This link doesn't work anymore 
```suggestion
- [Feature components](/uilib/extensions/forms/feature-fields) such as Email and PhoneNumber.
```

Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms.mdx]: Also, if we've "renamed" from `Feature components` to `Feature fields` we should take a quick search and replace, as I can still find occurrences of `Feature components` in the docs.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms.mdx [hunk 27]
```diff
@@ -1,20 +1,137 @@
...
+
+Forms is reusable components for data input, data display and surrounding layout for simplified user interface creation in React, built on top of base Eufemia components.
+
+The components constitute flexible building blocks that can be used individually, or together for a more practical data flow. They must also be easy to combine with components from other sources and local functionality in the individual application.
+
+The goal of the Forms extension is to make it easy to build forms and other data input features in DNB applications built with React.
+
+### Additional features
+
+- [Feature components](/uilib/extensions/forms/feature-fields) such as Email and PhoneNumber.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms.mdx:40]:
```suggestion
- [Feature fields](/uilib/extensions/forms/feature-fields) such as Email and PhoneNumber.
```

**Pull request merged**